spinn_front_end_common/interface/interface_functions/local_tdma_builder.py

```diff
@@ -118,69 +106,133 @@ def __call__(
+                    local_clocks, n_slots, clocks_between_phases,
+                    n_phases, clocks_per_cycle)
 
                 # test timings
-                self._test_timings(
-                    n_phases, time_between_phases, machine_time_step,
-                    time_scale_factor, fraction_of_sending, app_vertex.label)
+                fraction_of_sending = self._get_fraction_of_sending(
+                    n_phases, clocks_between_phases, clocks_for_sending)
+                if fraction_of_sending is not None:
+                    max_fraction_of_sending = max(
+                        max_fraction_of_sending, fraction_of_sending)
+
+        time_scale_factor_needed = time_scale_factor * max_fraction_of_sending
+        if max_fraction_of_sending > 1:
+            logger.warning(
+                "A time scale factor of {} may be needed to run correctly"
+                .format(time_scale_factor_needed))
+        elif max_fraction_of_sending < 1:
+            logger.info(
+                "The time scale factor could be reduced to {}"
+                .format(time_scale_factor_needed))
 
         # get initial offset for each app vertex.
         for app_vertex in application_graph.vertices:
             if isinstance(app_vertex, TDMAAwareApplicationVertex):
                 initial_offset = self._generate_initial_offset(
-                    app_vertex, app_verts, time_between_cores,
-                    machine_time_step, time_scale_factor, fraction_of_waiting)
+                    app_vertex, app_verts, clocks_initial,
+                    clocks_waiting)
                 app_vertex.set_initial_offset(initial_offset)
 
+    @staticmethod
+    def _auto_config_times(
+            app_machine_quantity, clocks_between_cores, clocks_for_sending,
+            app_vertex, n_keys_map, machine_graph, clocks_waiting):
+
+        n_cores = app_vertex.get_n_cores()
+        n_phases = app_vertex.find_n_phases_for(machine_graph, n_keys_map)
+
+        # If there are no packets sent, pretend there is 1 to avoid division
+        # by 0; it won't actually matter anyway
+        if n_phases == 0:
+            n_phases = 1
+
+        # overall time of the TDMA window minus initial offset
+        overall_clocks_available = clocks_for_sending - clocks_waiting
+
+        # easier bool compares
+        core_set = clocks_between_cores is not None
+        app_set = app_machine_quantity is not None
+
+        # adjust time between cores to fit time scale
+        if not core_set and app_set:
+            n_slots = int(math.ceil(n_cores / app_machine_quantity))
+            clocks_per_phase = (
+                int(math.ceil(overall_clocks_available / n_phases)))
+            # NOTE the plus 1 ensures the last core finishes, if its the worst
+            # in terms of n keys to transmit
+            clocks_between_cores = clocks_per_phase / (n_slots + 1)
+            logger.debug("adjusted clocks between cores is {}".format(
```

Comment:
Capital at start of comment.

Reply:
Fixed I think (along with a few others in the same method)